snapshots/manager.go

```diff
@@ -257,3 +290,44 @@ func (m *Manager) RestoreChunk(chunk []byte) (bool, error) {
 	}
 	return false, nil
 }
+
+// Snapshot takes a snapshot of the current state and prunes any old snapshottypes.
+func (m *Manager) Snapshot(height int64) {
+	if m == nil {
+		return
+	}
+	if !m.shouldTakeSnapshot(height) {
+		m.logger.Info("snapshot is skipped", "height", height) // TODO: change log level
+		return
+	}
+	go m.snapshot(height)
+}
+
+// shouldTakeSnapshot returns true is snapshot should be taken at height.
+func (m *Manager) shouldTakeSnapshot(height int64) bool {
+	return m.opts.Interval > 0 && uint64(height)%m.opts.Interval == 0
+}
```

Comment:
Just want to double-check, the idea is we take a snapshot at every snapshot height, but we delay prunings if the snapshot is still writing?

Reply:
That's correct. If the snapshot is on, we never prune the `snapshot-interval` heights. They get added to the `list to be pruned` after the snapshot is complete